### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by Josphat205**
### **In this Branch**

- Solved  the N+1 problem
- Used Capybara to write integration tests for each view
- Tested User index page
- Tested User show page
- Tested Post index page
- Tested Post show page

### CODE REVIEW

File: spec/features/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      visit users_path
+      expect(page).to have_xpath("//img[contains(@src,'http://hello.com/org.png')]")
+      expect(page).to have_xpath("//img[contains(@src,'http://hello.com/ogene.png')]")
+    end
+    it 'should redirect to users show page' do
+      visit users_path(@user.id)
+      click_on 'show user', match: :first
+      expect(page).to have_current_path user_path([@user.id + 1])
+    end
+  end
```
Comment (Felix45) [inline spec/features/user_index_spec.rb]: - [Required] Please consider adding a feature test that checks for the number of posts each user has written on the `users` index page as requested in the project requirements. Please see the screenshot below and rectify this.

## Project requirements
Screenshot from 2023-03-08 14-47-57

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+RSpec.describe 'User', type: :feature do
```
Comment (Felix45) [inline spec/features/user_show_spec.rb:4]: - [Required] Please consider adding a feature test that checks when a user clicks a post, it redirects the `post's` show page as requested in the project requirements. Please see the screenshot below and rectify this.

## Project requirements
Screenshot from 2023-03-08 14-52-39

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Josphat205,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [x] No linter errors :white_check_mark: 
- [x] All unit tests are passing :white_check_mark: 
- [x] Gitflow is used correctly :white_check_mark:  
- [x] Descriptive pull request title and summary :white_check_mark: 

## Required Changes ♻️
- [Required] Please consider adding a `LICENSE` file to your project and a link to the `LICENSE` on your readme file. when I click the License link on your readme file it redirects me to the `MIT` LIcense template, this link should lead to a `License` file in the root directory of your project. Please check on this, currently, it opens this Link [https://choosealicense.com/licenses/mit/](https://choosealicense.com/licenses/mit/) which should not be the case.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/user_show_spec.rb (hunk 2 above)
Comment (Josphat205) [reply spec/features/user_show_spec.rb:4]: sorted

File: spec/features/user_index_spec.rb (hunk 1 above)
Comment (Josphat205) [reply spec/features/user_index_spec.rb]: sorted

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi Team 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

soclose2


 ## Highlights
Nice UI design :heavy_check_mark: 
Great collaborative effort :heavy_check_mark: 

 ## Required Changes ♻️

- [ ] As requested by the previous reviewer, kindly work on updating the `License` file since it is an important part of your project. When the link to your license file is clicked, it leads to a placeholder MIT template file which is not customized. Kindly create a text file (typically named LICENSE or LICENSE.txt) in the root of your source code and copy the text of the license into the file. Replace [year] with the current year and [fullname] with the name (or names) of the copyright holders. :pray: 



_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @Josphat205 and @Krishnabot,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**